tests/unit_tests/PreprocessorTests.cpp

```diff
@@ -194,7 +194,7 @@ TEST_CASE("Function macro (keyword as formal argument)", "[preprocessor]") {
 }
 
 TEST_CASE("Macro pasting (identifiers)", "[preprocessor]") {
-    auto& text = "`define FOO(x,y) x   ``   _blah``y\n`FOO(   bar,    _BAZ)";
+    auto& text = "`define FOO(x,y) x``_blah``y\n`FOO(   bar,    _BAZ)";
```

Comment:
So after looking at the spec again, I agree that the paste operator should work this way. It's less of a paste operator like in C and more just some placeholder token to separate other tokens.

With that said, I think this change is more complicated than it needs to be. In the loop to handle macro expansion, you can check if the paste operator token and the rhs token both have .trivia().empty(). If not, then you just ignore the operator (maybe we'll issue a warning or something). That way we don't need a new dummy token type.